{
  "guid": "60a71766-b0d6-4d8e-9223-8d5628c1bec0",
  "name": "Chapter 5 - Error Handling",
  "categories": [ "Toolkit/Publication: Data Parallel C++" ],
  "description": "Chapter 05 source code examples: Error Handling | fig_5_1_async_task_graph.cpp * Figure 5-1. Separation of host program and task graph executions | fig_5_2_sync_error.cpp * Figure 5-2. Creating a synchronous error | fig_5_3_async_error.cpp * Figure 5-3. Creating an asynchronous error | fig_5_4_unhandled_exception.cpp * Figure 5-4. Unhandled exception in C++ | fig_5_5_terminate.cpp * Figure 5-5. std::terminate is called when a SYCL asynchronous exception isnâ€™t handled | fig_5_6_catch_snip.cpp * Figure 5-6. Pattern to catch sycl::exception specifically | fig_5_7_catch.cpp * Figure 5-7. Pattern to catch exceptions from a block of code | fig_5_8_lambda_handler.cpp * Figure 5-8. Example asynchronous handler implementation defined as a lambda | fig_5_9_default_handler_proxy.cpp *	Figure 5-9. Example of how the default asynchronous handler behaves ",
  "toolchain": [ "dpcpp" ],
  "languages": [ { "cpp": {} } ],
  "targetDevice": [ "CPU", "GPU" ],
  "os": [ "linux", "windows" ],
  "builder": [ "ide", "cmake" ],
  "commonFolder": {
    "base": "..",
    "include": [
      "Ch05_error_handling",
      "LICENSE.txt",
      "README.md",
      "CMakeLists.txt"
    ],
    "exclude": [
      "sample-folder-name/test.sh"
    ]
   },
   "ciTests": {
    "linux": [
      {
        "steps": [
          "mkdir build",
          "cd build",
          "cmake ..",
          "make",
          "make run"
        ]
      }
    ],
    "windows": [
      {
        "steps": [
          "mkdir build",
          "cd build",
          "cmake -G Ninja ..",
          "ninja"
        ]
      }
    ]
  }
}